sharding/networksharding/peerShardMapper.go

```diff
@@ -313,7 +313,7 @@ func (psm *PeerShardMapper) UpdatePeerIdShardId(pid core.PeerID, shardId uint32)
 
 // UpdatePeerIdSubType updates the peerIdSubType search map containing peer IDs and peer subtypes
 func (psm *PeerShardMapper) UpdatePeerIdSubType(pid core.PeerID, peerSubType core.P2PPeerSubType) {
-	psm.peerIdSubType.HasOrAdd([]byte(pid), uint32(peerSubType), uint32Size)
+	psm.peerIdSubTypeCache.HasOrAdd([]byte(pid), peerSubType, uint32Size)
```

Comment:
why `HasOrAdd` if returned values are ignored? Or is it because of recent-ness?

Reply:
HasOrAdd adds the value if that value exists. It does the checking and adding atomically because this method can be called in a concurrent manner.